src/pgduckdb_duckdb.cpp

```diff
@@ -42,7 +53,21 @@ DuckDBManager::Initialize() {
 	auto connection = duckdb::make_uniq<duckdb::Connection>(*database);
 
 	auto &context = *connection->context;
-	context.Query("ATTACH DATABASE 'pgduckdb' (TYPE pgduckdb)", false);
+	auto &db_manager = duckdb::DatabaseManager::Get(context);
+	default_dbname = db_manager.GetDefaultDatabase(context);
+	pgduckdb::DuckDBQueryOrThrow(context, "ATTACH DATABASE 'pgduckdb' (TYPE pgduckdb)");
+	pgduckdb::DuckDBQueryOrThrow(context, "ATTACH DATABASE ':memory:' AS pg_temp;");
+
+	if (pgduckdb::IsMotherDuckEnabled()) {
+		/*
+		 * Workaround for MotherDuck catalog sync that turns off automatically,
+		 * in case of no queries being sent to MotherDuck. Since the background
+		 * worker never sends any query to MotherDuck we need to turn this off.
+		 * So we set the timeout to an arbitrary very large value.
+		 */
+		pgduckdb::DuckDBQueryOrThrow(context,
+		                             "SET motherduck_background_catalog_refresh_inactivity_timeout='99 years'");
```

Comment:
The good old 99y bug 😆

Reply:
Yeah, I'm pretty certain no-one runs their postgres server without restart for that long. And even they do, it's not going to be my problem anymore.